From c0da3b51487fbe2b27b52263eec75c483b8e42d0 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:29:46 -0400
Subject: [net] openvswitch: Use net_ratelimit in OVS_NLERR

Message-id: <acc628b7622870b9ea4562b0ddb91a0aeb9d4ec9.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94022
O-Subject: [RHEL7.1 net PATCH 13/50] openvswitch: Use net_ratelimit in OVS_NLERR
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 1815a8831fb04c60d73627816cb0b596266e9bee
Author: Joe Perches <joe@perches.com>
Date:   Mon Feb 3 17:06:46 2014 -0800

    openvswitch: Use net_ratelimit in OVS_NLERR

    Each use of pr_<level>_once has a per-site flag.

    Some of the OVS_NLERR messages look as if seeing them
    multiple times could be useful, so use net_ratelimit()
    instead of pr_info_once.

    Signed-off-by: Joe Perches <joe@perches.com>
    Signed-off-by: Jesse Gross <jesse@nicira.com>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/datapath.h b/net/openvswitch/datapath.h
index 0531738..7ede507 100644
--- a/net/openvswitch/datapath.h
+++ b/net/openvswitch/datapath.h
@@ -194,7 +194,9 @@ struct sk_buff *ovs_vport_cmd_build_info(struct vport *, u32 pid, u32 seq,
 int ovs_execute_actions(struct datapath *dp, struct sk_buff *skb);
 void ovs_dp_notify_wq(struct work_struct *work);
 
-#define OVS_NLERR(fmt, ...) \
- pr_info_once("netlink: " fmt, ##__VA_ARGS__)
-
+#define OVS_NLERR(fmt, ...)     \
+do {        \
+ if (net_ratelimit())     \
+  pr_info("netlink: " fmt, ##__VA_ARGS__); \
+} while (0)
 #endif /* datapath.h */
-- 
1.7.1